pkg/operation/botanist/health_check.go

```diff
@@ -279,6 +283,102 @@ func CheckSystemComponents(
 	return nil, nil
 }
 
+// checkAPIServerAvailability checks if the API server of a Shoot cluster is reachable and measure the response time.
+func (b *Botanist) checkAPIServerAvailability(condition *gardenv1beta1.Condition) *gardenv1beta1.Condition {
+	// Try to reach the Shoot API server and measure the response time.
+	now := time.Now()
+	response, err := b.K8sShootClient.Curl("/healthz")
+	responseDurationText := fmt.Sprintf("[response_time:%dms]", time.Now().Sub(now).Nanoseconds()/time.Millisecond.Nanoseconds())
+	if err != nil {
+		message := fmt.Sprintf("Request to Shoot API server failed. %s (%s)", responseDurationText, err.Error())
+		return helper.UpdatedCondition(condition, corev1.ConditionFalse, "RequestFailed", message)
+	}
+
+	// Determine the status code of the response.
+	var statusCode int
+	response.StatusCode(&statusCode)
+
+	if statusCode != http.StatusOK {
+		var body string
+		bodyRaw, err := response.Raw()
+		if err != nil {
+			body = fmt.Sprintf("Could not parse response body: %s", err.Error())
+		} else {
+			body = string(bodyRaw)
+		}
+		message := fmt.Sprintf("Shoot API server health check returned a non ok status code %d. %s (%s)", statusCode, responseDurationText, body)
+		return helper.UpdatedCondition(condition, corev1.ConditionFalse, "ResponseCodeError", message)
+	}
+
+	message := fmt.Sprintf("Shoot API server responded with success status code. %s", responseDurationText)
+	return helper.UpdatedCondition(condition, corev1.ConditionTrue, "ResponseCodeOK", message)
+}
```

Comment:
Do you want to keep it this way or change it to several times?

Reply:
I have no strong opinion about this. I was just curious because @dkistner explained to me that the condition will be only red if multiple checks fail, however, this implementation does not cover it. I just wanted to know what you plan :)

Reply:
We will continue with the current implementation for now (red immediately on first failure). If you want to change it to the intended behaviour you explained to me then you can file a separate PR in the future.
Still, I'm curious about what's planned here.